lib/widgets/selection.dart

```diff
@@ -1671,6 +1672,40 @@ class _AddToPlaylistSelectionAction extends StatelessWidget {
   }
 }
 
+/// Action that removes a song from playlist.
+class _FavoriteSelectionAction extends StatelessWidget {
+  const _FavoriteSelectionAction({ Key? key }) : super(key: key);
+
+  void _handleTap(BuildContext context, ContentSelectionController controller) {
+    final contentTuple = ContentUtils.selectionPack(controller.data);
+    FavoritesControl.instance.setFavorite(
+      contentTuple: contentTuple,
+      value: contentTuple.any((el) => !el.isFavorite),
+    );
+    controller.close();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final l10n = getl10n(context);
+    final controller = ContentSelectionController._of(context);
+    final theme = ThemeControl.instance.theme;
+    return _ActionBuilder(
+      controller: controller,
+      shown: () => true,
+      builder: (context, child) => EmergeAnimation(
+        animation: controller.animation,
+        child: HeartButton(
+          tooltip: l10n.addToFavorites,
+          active: controller.data.any((el) => !el.data.isFavorite),
```

Comment:
When executing the action, the button starts to animate. This feels a bit weird, because it changes to a hollow heart when you just favored something:
![_device-2022-05-15-004739](https://user-images.githubusercontent.com/6966049/168450464-ccb83554-1b15-4fab-bc8b-df674d6e1510.gif)

Reply:
Fixed it, thanks!